File: Java/CodesInterviews/src/main/java/org/example/arrays/ContainsDuplicate.java
package org.example.arrays;

import java.util.HashSet;
import java.util.Set;

public class ContainsDuplicate {

    public boolean containsDuplicate(int[] nums) {

        if (nums.length == 0) return false;

        Set<Integer> set = new HashSet<>();
        for (int num : nums) {
            if (set.contains(num)) {
                return true;
            } else {
                set.add(num);
            }
        }
        return false;
    } // TC: O(n), SC: O(n)

    public static void main(String[] args) {
        int[] nums = {3, 4, 5, 5, 3, 1, 2};
        ContainsDuplicate c = new ContainsDuplicate();
        System.out.println(c.containsDuplicate(nums));
    }
}

File: Kotlin/CodesInterviews/src/main/kotlin/arrays/ContainsDuplicate.kt
package arrays
class ContainsDuplicate {

    fun containsDuplicate(nums: IntArray): Boolean {
        if (nums.isEmpty()) return false

        val set: MutableSet<Int> = mutableSetOf()
        for (num in nums) {
            if (set.contains(num)) {
                return true
            } else {
                set.add(num)
            }
        }
        return false
    }
}

fun main() {
    val duplicate = ContainsDuplicate()
    val nums = intArrayOf(1,1,1,3,3,4,3,2,4,2)
    println(duplicate.containsDuplicate(nums))
}